test/images/devenv/run.sh

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The pipelines-service Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+usage() {
+    echo "
+Usage:
+    ${0##*/} [options]
+
+Run a container with the development environment.
+
+Optional arguments:
+    -d, --debug
+        Activate tracing/debug mode.
+    -h, --help
+        Display this message.
+
+Example:
+    ${0##*/}
+" >&2
+}
+
+init() {
+    SCRIPT_DIR=$(
+        cd "$(dirname "$0")" >/dev/null
+        pwd
+    )
+    PROJECT_DIR=$(
+        cd "$SCRIPT_DIR" >/dev/null
+        git rev-parse --show-toplevel
+    )
+}
+
+parse_args() {
+    while [[ $# -gt 0 ]]; do
+        case $1 in
+        -d | --debug)
+            set -x
+            ;;
+        -h | --help)
+            usage
+            exit 0
+            ;;
+        *)
+            echo "Unknown argument: $1"
+            usage
+            exit 1
+            ;;
+        esac
+        shift
+    done
+}
+
+get_image_name() {
+    DEPENDENCIES_SHA=$(
+        cat "$SCRIPT_DIR/Dockerfile" "$SCRIPT_DIR/install.sh" "$PROJECT_DIR/config/dependencies.yaml" |
+            sha256sum | cut -c1-8
+    )
+    IMAGE_NAME="pipeline-service/devenv:$DEPENDENCIES_SHA"
+}
+
+build_image() {
+    if ! podman images --filter "label=name=$IMAGE_NAME" | grep -q "localhost/"; then
+        echo "[Building container]"
+        podman build -f "$SCRIPT_DIR/Dockerfile" --label "name=$IMAGE_NAME" -t "$IMAGE_NAME" "$PROJECT_DIR"
+    fi
+}
+
+get_container_name() {
+    CONTAINER_NAME=$(pwd | sed -e 's:/:__:g' -e 's:[^a-zA-Z0-9_-]:-:g' | cut -c3-)
+}
+
+start_container() {
+    if ! podman ps --filter "label=name=$IMAGE_NAME" | grep -q "$CONTAINER_NAME"; then
```

Comment:
if we are matching with filter label which has a SHA associated with it, then why do we need to grep it with the container name?

Reply:
I think devcontainers can work with the toolbox too.

Reply:
You might have the same image used to spawn different containers. E.g you've clone the project in multiple directories to work on features/fixes in parallel.